paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc

```diff
@@ -497,23 +504,19 @@ static void ExecuteMatMul(const ExecutionContext& ctx) {
   auto* x = ctx.Input<Tensor>("X");
   auto* y = ctx.Input<Tensor>("Y");
   auto* out = ctx.Output<Tensor>("Out");
-  float alpha = ctx.HasAttr("alpha") ? ctx.Attr<float>("alpha") : 1.0f;
   const auto& dev_ctx =
       ctx.template device_context<paddle::platform::MKLDNNDeviceContext>();
+  const auto& onednn_engine = dev_ctx.GetEngine();
```

Comment:
Do you need to get engine here?
Since ctx is passed to MatMulMKLDNNHandler as an attribute it can be accessed from there.

Reply:
Yes, but engine is passed directly to constructor, so I think it's cleaner to create object and just pass it as argument